src/libs/actions/PersonalDetails.ts

```diff
@@ -296,7 +297,7 @@ function updateAutomaticTimezone(timezone: Timezone) {
     };
 
     const parameters: UpdateAutomaticTimezoneParams = {
-        timezone: JSON.stringify(timezone),
+        timezone: JSON.stringify(DateUtils.parseTimezone(timezone)),
```

Comment:
Looks like you have not updated the timezone saved in Onyx.

https://github.com/Expensify/App/pull/33186/files#diff-1372b8700afa9b26d89dfe0575dcfaaf399d62b9b0555f30863e8c19c5609df9R310